numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -292,6 +343,205 @@ def test_fma_f8(self):
         compiled[1, 1](ary, 2., 3., 4.)
         np.testing.assert_allclose(ary[0], 2 * 3 + 4)
 
+    def toolkit_supported(self, toolkit_major, toolkit_minor):
+        return cuda_X_or_above(toolkit_major, toolkit_minor)
+
+    def compute_supported(self, compute_major, compute_minor):
+        return cc_X_or_above(compute_major, compute_minor)
+
+    def test_hadd(self):
+        # 16-bit Floating point binary operations add, sub, and mul were
+        # introduced in PTX ISA 4.2 (Cuda 7.0) and SM 5.3
+
+        if self.compute_supported(5,3) and self.toolkit_supported(7,0):
+            compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hadd)
+            ary = np.zeros(1, dtype=np.float16)
+            arg1 = np.array([3.], dtype=np.float16)
+            arg2 = np.array([4.], dtype=np.float16)
+            compiled[1, 1](ary, arg1, arg2)
+            np.testing.assert_allclose(ary[0], arg1 + arg2)
+        else:
+            self.skipTest("Test requires toolkit >= 7.0 and SM >= 5.3")
+
+    def test_hadd_scalar(self):
+        if self.compute_supported(5,3) and self.toolkit_supported(7,0):
+            compiled = cuda.jit("void(f2[:], f2, f2)")(simple_hadd_scalar)
+            ary = np.zeros(1, dtype=np.float16)
+            arg1 = np.float16(3.1415926)
+            arg2 = np.float16(3.)
+            compiled[1, 1](ary, arg1, arg2)
+            ref = arg1 + arg2
+            np.testing.assert_allclose(ary[0], ref)
+        else:
+            self.skipTest("Test requires toolkit >= 7.0 and SM >= 5.3")
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_hadd_ptx(self):
+        if self.compute_supported(5,3) and self.toolkit_supported(7,0):
```

Comment:
No need to check for toolkit 7.0 here.

Reply:
Also, because we're only compiling PTX for this test, we don't need a device that supports it - `compile_ptx` takes an argument for the compute .capability to use, so instead of skipping these tests, we can always run them and pass `cc=(5, 3)` to `compile_ptx()`: https://numba.readthedocs.io/en/latest/cuda-reference/host.html#numba.cuda.compile_ptx